utils/scripts/setup-env.bash

```diff
@@ -0,0 +1,129 @@
+#!/bin/bash
+
+# This script sets up the lingua franca environment; right now, it only works on Debian forks.
+
+set -euo pipefail
+
+CROW_URL="https://github.com/CrowCpp/Crow/releases/download/v1.0%2B5/crow-v1.0+5.deb"
+
+# Check if running Debian fork; the whole script requires apt to properly run.
+if [ ! -f "/etc/debian_version" ]; then
+    echo "This script requires apt to run, but it appears that this distro is not a debian fork." 1>&2;
+    exit 1
+fi
+
+SETUP_NETWORK=true
+SETUP_ROS=true
+ROS_USE_OFFICIAL_REPO=true
+INSTALL_USER_DEPENDENCIES=false
+
+for arg in "$@"; do
+    shift
+    case "$arg" in
+        '--install-user-dependencies') INSTALL_USER_DEPENDENCIES=true;;
+        '--no-setup-network') SETUP_NETWORK=false;;
+        '--no-setup-ros') SETUP_ROS=false;;
+        '--ros-no-use-official-repo') ROS_USE_OFFICIAL_REPO=false;;
+        '--debug') set -x;; 
+    esac
+done
+
+# Install dependencies
+sudo apt-get update
+## Setup C, C++, Python, Rust, protobuf, gRPC
+sudo apt-get install --assume-yes \
+    build-essential \
+    python3 python3-dev python3-pip \
+    rustc cargo
+sudo python3 -m pip install --upgrade pip
+# Install python dependencies and
+# latest CMake; see https://www.kitware.com/cmake-python-wheels/ https://askubuntu.com/a/1070770
+sudo python3 -m pip install --exists-action i requests setuptools cmake
+
+if [ $SETUP_NETWORK = true ]; then 
+    # Setup protobuf, libgrpc, mqtt, and libwebsockets; libwebsockets-dev installs libwebsockets16 or 17
+    sudo apt-get install --assume-yes \
+        libprotobuf-dev libprotobuf-c-dev protobuf-c-compiler protobuf-compiler python3-protobuf \
+        protobuf-compiler-grpc libgrpc-dev libgrpc++-dev \
+        libwebsockets-dev libpaho-mqtt-dev
+    # Setup crow
+    curl --proto '=https' --tlsv1.2 -L -o ./crow.deb "${CROW_URL}"
```

Comment:
As this is intended to be run by users on their system, we should at a checksum check. Otherwise, someone could install arbitrary code with a man in the middle attack.

Reply:
A checksum is a good idea, but this is not meant to be run by users in their own machines.

Reply:
Even if users wants to run it, (I think) curl with tls could largely prevent MITM already?